### PR TITLE
[R-package] fix CVBooster reset_parameter() method (fixes #4900)

### LINKED ISSUE #4900: [R-package] CVBooster$reset_parameter() is broken
## Description

`lgb.cv()` returns an object of class `CVBooster`. That class has a public method, `$reset_parameter()`, which is intended to be used to update the parameters of each `Booster` in `CVBooster$boosters`.

That method, which was added about 5 years ago (https://github.com/microsoft/LightGBM/commit/f3f2f5a9539660eef54b8eec5236fb926923d015), currently throws an error when called.

> Error: object 'boosters' not found

## Reproducible example

```r
library(lightgbm)
data("agaricus.train")
dtrain <- lightgbm::lgb.Dataset(
    data = agaricus.train$data
    , label = agaricus.train$label
)
cv_bst <- lightgbm::lgb.cv(
    params = list(
        num_leaves = 8L
    )
    , nfold = 3L
    , nrounds = 10L
    , data = dtrain
    , obj = "binary"
    , verbose = -1L
)
cv_bst$reset_parameter(
    list(num_leaves = 11L)
)
```

> Error in cv_bst$reset_parameter(list(num_leaves = 11L)) : 
  object 'boosters' not found

## Environment info

LightGBM version or commit hash: v3.3.1.99 (https://github.com/microsoft/LightGBM/commit/7b10baffb6e600862b087a4207343b0b58eed2b5)

Command(s) you used to install LightGBM

```shell
sh build-cran-package.sh --no-build-vignettes && \
R CMD INSTALL ./lightgbm_3.3.1.99.tar.gz
```

Using R 4.1.0 on macOS.

## Additional Comments

In the Python package, this works without issue.

```python
import lightgbm as lgb
from sklearn.datasets import make_regression

X, y = make_regression(n_samples=10_000, n_features=10, n_informative=5)

dtrain = lgb.Dataset(data=X, label=y)

results = lgb.cv(
    train_set=dtrain,
    params={
        "objective": "regression_l2",
        "num_leaves": 8,
        "verbose": -1
    },
    stratified=False,
    shuffle=False,
    return_cvbooster=True
)
cv_bst = results["cvbooster"]
for bst in cv_bst.boosters:
    print(bst.params)

# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}

_ = cv_bst.reset_parameter({"num_leaves": 11})
for bst in cv_bst.boosters:
    print(bst.params)

# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
# {'objective': 'regression_l2', 'num_leaves': 11, 'verbose': -1, 'num_iterations': 100, 'early_stopping_round': None}
```

### PR DESCRIPTION
**Pull request by jameslamb**
Fixes #4900.

### CODE REVIEW

File: R-package/tests/testthat/test_basic.R [hunk 1]
```diff
@@ -380,6 +380,30 @@ test_that("cv works", {
...
+      , min_data = 1L
+      , num_leaves = 7L
+      , verbose = VERBOSITY
+    )
+    , data = dtrain
+    , nrounds = 2L
+    , nfold = 2L
+  )
+  expect_is(cv_bst, "lgb.CVBooster")
+  expect_length(cv_bst$boosters, 2L)
```
Comment (StrikerRUS) [inline R-package/tests/testthat/test_basic.R:397]: Please use different values for `nrounds` and `nfold` arguments to make it clearer what value is asserted in `expect_length(cv_bst$booster, ...`.

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
Thanks! Just one small comment in test.

File: R-package/tests/testthat/test_basic.R (hunk 1 above)
Comment (jameslamb) [reply R-package/tests/testthat/test_basic.R]: ah yeah, good idea!

updated in https://github.com/microsoft/LightGBM/pull/4901/commits/52bce82c4a7ad8b13e531956b7f23f3cadb9b02c

**Review by StrikerRUS — APPROVED** [review]
LGTM!

**Pull request merged**